### PR TITLE
feat(search): add youtube search

### PR DESCRIPTION
**Pull request by OCannings**
A first attempt at integrating the Youtube search API has been made.
This is a work in progress and currently untested.

An environment variable has been introduced: YOUTUBE_API_KEY

Visit http://localhost:4000/api/videos?q=hello to see a list of videos
pulled from Youtube.

TODO: write automated tests and handle API failures/malformed JSON
responses

### CODE REVIEW

File: config/dev.exs [hunk 1]
```diff
@@ -41,3 +41,6 @@ config :juicebox, Juicebox.Repo,
   database: "juicebox_dev",
   hostname: "localhost",
   pool_size: 10
+
+config :juicebox, Juicebox.Youtube,
+  api_key: ""
```
Comment (ljkbennett) [inline config/dev.exs]: why is the dev version empty?

Comment (OCannings) [inline config/dev.exs]: I just removed my api key before committing. Need something better here.

**Comment by ljkbennett** [comment]
I set up an API key and added it to dev.exs directly.
Getting an error message: `protocol Enumerable not implemented for nil` on line 7 of api_controller.ex

**Comment by OCannings** [comment]
@littleowllabs Looks like you have an invalid API key. I just updated the pr to handle that case.

**Comment by ljkbennett** [comment]
Just worked it out. I had a valid API key, but I hadn't enabled the API 😿

File: config/dev.exs [hunk 2]
```diff
@@ -41,3 +41,8 @@ config :juicebox, Juicebox.Repo,
   database: "juicebox_dev",
   hostname: "localhost",
   pool_size: 10
+
+config :juicebox, :youtube_api, Juicebox.Youtube.HTTP
+
+config :juicebox, Juicebox.Youtube,
+  api_key: ""
```
Comment (ljkbennett) [inline config/dev.exs]: why not put the api key into an environment variable for dev too?

File: lib/juicebox/youtube/http.ex [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+      "snippet" => %{
+        "title" => title,
+        "description" => description,
+        "thumbnails" => %{
+          "high" => %{
+            "url" => thumbnail
+          }
+        }
+      }
+    } = video
```
Comment (ljkbennett) [inline lib/juicebox/youtube/http.ex:59]: Is this the preferred approach rather than doing it in the definition when there is a big pattern to match?

Comment (OCannings) [inline lib/juicebox/youtube/http.ex:59]: Like this?

``` elixir
defp format_video(%{
      "id" => %{
        "videoId" => video_id
      },
      "snippet" => %{
        "title" => title,
        "description" => description,
        "thumbnails" => %{
          "high" => %{
            "url" => thumbnail
          }
        }
      }
    }) do

    %Video{
       video_id: video_id,
       title: title,
       description: description,
       thumbnail: thumbnail
     }
  end
```

Comment (ljkbennett) [inline lib/juicebox/youtube/http.ex:59]: Yeah, was just thinking aloud really. I think it's ugly, but it'd not much less ugly than the other way and saves extra intermediate allocations.

**Pull request merged**